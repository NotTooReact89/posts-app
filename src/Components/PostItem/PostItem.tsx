import {
  Avatar,
  Button,
  createStyles,
  Grid,
  makeStyles,
  Paper,
  Theme,
  Typography,
} from "@material-ui/core";
import ModeCommentIcon from "@material-ui/icons/ModeComment";
import React from "react";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      overflow: "hidden",
      padding: theme.spacing(0, 3),
    },
    paper: {
      maxWidth: 400,
      margin: `${theme.spacing(1)}px auto`,
      padding: theme.spacing(2),
    },
    button: {
      marginTop: theme.spacing(1),
      marginBottom: theme.spacing(1),
    },
    small: {
      fontSize: "1rem",
    },
  })
);

interface ICommentItem {
  postId: string;
  id: string;
  name: string;
  email: string;
  body: string;
}

interface IPostItem {
  title?: string;
  body: string;
  initials: string;
  name: string;
  comments?: Array<ICommentItem>;
  onClick?: () => void;
  isComment?: boolean;
}

const PostItem = ({
  title,
  body,
  initials,
  name,
  comments,
  onClick,
  isComment = false,
}: IPostItem) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Paper className={classes.paper}>
        <Grid container direction="row" alignItems="center">
          <Avatar
            style={{
              marginRight: 10,
            }}
            className={classes.small}
          >
            {initials}
          </Avatar>
          {" " + name}
        </Grid>
        <Grid container direction="column" wrap="nowrap" spacing={2}>
          <Grid item xs>
            <Typography variant="h6">{title}</Typography>
          </Grid>
          <Grid item xs>
            <Typography>{body}</Typography>
          </Grid>
        </Grid>
        {!isComment && (
          <Grid container direction="row" alignItems="center">
            <Button
              variant="contained"
              className={classes.button}
              startIcon={<ModeCommentIcon />}
              onClick={onClick}
            >
              {`${comments?.length} comments`}
            </Button>
          </Grid>
        )}
      </Paper>
    </div>
  );
};

export default PostItem;
