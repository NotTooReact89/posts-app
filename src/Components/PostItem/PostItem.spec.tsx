import TestRenderer from "react-test-renderer";
import PostItem from "./PostItem";

describe("PostItem", () => {
  it("should render the component", () => {
    const component = TestRenderer.create(
      <PostItem name="abc" body="xyz" initials="T" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
