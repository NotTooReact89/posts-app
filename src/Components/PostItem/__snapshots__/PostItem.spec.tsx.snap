// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem should render the component 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-small-4 MuiAvatar-colorDefault"
        style={
          Object {
            "marginRight": 10,
          }
        }
      >
        T
      </div>
       abc
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          xyz
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18z"
              />
            </svg>
          </span>
          undefined comments
        </span>
      </button>
    </div>
  </div>
</div>
`;
